plugins/serializers/json/json.go

```diff
@@ -4,17 +4,19 @@ import (
 	ejson "encoding/json"
 
 	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
 )
 
 type JsonSerializer struct {
+	TimestampUnits internal.Duration
 }
 
 func (s *JsonSerializer) Serialize(metric telegraf.Metric) ([]byte, error) {
 	m := make(map[string]interface{})
 	m["tags"] = metric.Tags()
 	m["fields"] = metric.Fields()
 	m["name"] = metric.Name()
-	m["timestamp"] = metric.UnixNano() / 1000000000
+	m["timestamp"] = metric.UnixNano() / s.TimestampUnits.Duration.Nanoseconds()
```

Comment:
Guard against the duration being 0s or negative, just emit them at the default units.